kore/src/Kore/Step/Simplification/Not.hs

```diff
@@ -174,11 +174,25 @@ makeTermNot
 -- TODO: maybe other simplifications like
 -- not ceil = floor not
 -- not forall = exists not
+makeTermNot (Not_ _ term) = term
```

Comment:
`term` might actually be an `Internal.Predicate` collapsed to a `Syntax.Predicate` in `makeEvaluatePredicate`. This function should return a simplified thing, that can't be simplified further, but simplifying the result again would produce the original `Internal.Predicate`